server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -136,6 +136,11 @@ public Analyzer getAnalyzer(String analyzer) throws IOException {
                             throw new ElasticsearchException("failed to load analyzer for name " + key, ex);
                         }}
             );
```

Comment:
Do we need this part?  I think just having the deprecation message in the AnalyzerProvider should be enough?

Reply:
I think this part for this line.
https://github.com/elastic/elasticsearch/blob/76f6f09e2ea976c25312b45815b98e780173f8c9/modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java#L326
If Es find `standard_html_strip` from `prebuiltAnalysis`, Es doesn't use `StandardHtmlStripAnalyzerProvider`.

Reply:
Ah, ok.  Can you move the deprecation message into the PreBuiltAnalyzerProviderFactory?  Like https://github.com/elastic/elasticsearch/blob/76f6f09e2ea976c25312b45815b98e780173f8c9/modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java#L440
I'd rather not have specific analyzers referred to in the AnalysisRegistry if we can avoid it.

Reply:
Ah, I pushed it but it failed.
I think the timing to call `create` function is different between `PreBuiltAnalyzerProviderFactory` and `PreConfiguredTokenFilter`.
And I just remembered why I added check logic in AnalysisRegistry...

Reply:
+1, thanks!

Reply:
@romseygeek Thanks! I will revert last commit, then will merge it.